datumaro/plugins/voc_format/converter.py

```diff
@@ -201,7 +201,7 @@ def save_subsets(self):
                     ET.SubElement(source_elem, 'annotation').text = 'Unknown'
                     ET.SubElement(source_elem, 'image').text = 'Unknown'
 
-                    if item.has_image:
+                    if item.has_image and item.image.has_data:
```

Comment:
Image can have size info even when no image is available. Use `image.has_size` instead.